kiosk/src/Components/EnterHighScore.tsx

```diff
@@ -11,31 +11,87 @@ interface IProps {
   }
 
 const EnterHighScore: React.FC<IProps> = ({ kiosk }) => {
-    const [initialIndexRef, ] = useState(new PrimitiveRef(0));
-    const [initialIndex, setInitialIndex] = useState(initialIndexRef.value);
+    const [indexChanged, setIndexChanged] = useState(false);
     const [initials, setInitials] = useState(Array(configData.HighScoreInitialsLength + 1).join(configData.HighScoreInitialAllowedCharacters[0]));
+    const [timesAPressed, setTimesAPressed] = useState(0);
+    const [runOnce, setRunOnce] = useState(false);
+    const [firstRun, setFirstRun] = useState(true);
+
+    const existingHighScores = kiosk.getHighScores(kiosk.selectedGame!.id);
+    const aboveScores = existingHighScores.filter(item => item.score > kiosk.mostRecentScores[0]);
+    const belowScores = existingHighScores.slice(aboveScores.length, existingHighScores.length);
 
     if (!kiosk.mostRecentScores || !kiosk.mostRecentScores.length) {
         throw new Error("Cannot load high score entry view without having recent scores");
     }
 
-    const renderList = (highScores: HighScore[], offset: number): JSX.Element[] => {
-        return highScores.map((highScore, i) => {
-            return (
-                <tr className="highScoreRow" key={i}>
-                    <td className="highScoreIndex">{i + offset}.</td>
-                    <td className="highScoreInitials">
-                        <span className="highScoreInitial">{highScore.initials[0]}</span>
-                        <span className="highScoreInitial">{highScore.initials[1]}</span>
-                        <span className="highScoreInitial">{highScore.initials[2]}</span>
-                    </td>
-                    <td className="highScoreInitialsScoreSpacer"></td>
-                    <td className="highScoreScore">{highScore.score}</td>
-                </tr>
-            )
-        })
+    const gamepadLoop = () => {
+        if (kiosk.state !== KioskState.EnterHighScore) { return; }
+
+        if (kiosk.gamepadManager.isAButtonPressed()) {
+            setIndexChanged(true);
+        } else {
+            setIndexChanged(false);
+        }
+
+        if (kiosk.gamepadManager.isBButtonPressed()) {
+            kiosk.navigate(KioskState.GameOver);
+        }
+    };
+
+    const lookForPressed = () => {
+        let interval: any;
+        let timeout: any;
+        timeout = setTimeout(() => {
+            interval = setInterval(() =>
+            gamepadLoop(),
+            configData.EnterHighScorePoll
+        )
+        }, configData.EnterHighScoreDelayMilli)
+
+
+        return () => {
+            if (interval) {
+                clearInterval(interval);
+            }
+            if (timeout) {
+                clearTimeout(timeout);
+            }
+        }
+    }
+
+    lookForPressed();
```

Comment:
this should probably be a useeffect and not just created as an anonymous function? the returned value here is never gonna be used / the interval won't be cleared, and react can rerun this function at any time if other state has changed (/ makes no guarantee how many times a component function will be called)